### ISSUE TITLE
Messages sent from DeskSMS appearing twice. (Nexus 5)

### ISSUE THREAD

**Issue by ghost** [ClockworkMod/DeskSMS#104]
Message first appears as "Delivery pending.." then appears again as a successfully sent message. The pending message does not disappear though and just hangs indefinitely.

![ef8c4748-8093-11e3-9335-e3641ceabb53](https://f.cloud.github.com/assets/5743203/1997747/c334da6c-8529-11e3-9e32-2da89ee840dc.png)
![e30c5b92-7b21-11e3-9adc-ee74aa8d739c](https://f.cloud.github.com/assets/5743203/1997748/c335f014-8529-11e3-8851-13b1fced79ff.png)

Reported by three other users here: https://github.com/ClockworkMod/DeskSMS/issues/77#issuecomment-31989824

I'm going to flash a bunch of crap to see if I can accidentally fix it.

**Comment by prasta** [comment]
I am experiencing this issue as well, but I don't believe it's related to the phone. It seems to be the webapp/chrome extension. I tried clearing synced data, and starting fresh, but I get the same result.

![screenshot at feb 20 08-01-28](https://f.cloud.github.com/assets/2895959/2218466/27d6c556-9a2f-11e3-83ce-6b4d74eecfb4.png)

**Comment by PGonza02** [comment]
Hi,
I'm having the same issue and I thought it was related to getting a new phone and the service still trying to send messages through the old phone as well as the new one.

**Comment by guersbru** [comment]
Hello,
Same issue with Nexus 4 not rooted.
SMS are properly sent but the webapp and extension display the same message twice (one with error other successfull). Message are displayed splitted when they exceed a certain number or characters.
![2014-03-31_163227](https://cloud.githubusercontent.com/assets/7116288/2568345/63cc4bbe-b8e1-11e3-83bf-83a8ccdf3e8e.jpg)

**Comment by pauloschilling** [comment]
@koush @ClockworkMod

Same issue here, Galaxy S4 international, Android 4.4.2, not rooted, default Samsung's firmware.

Also, long messages started to be sent as small individual messages, breaking in the middle of the words. If I send the same message directly from my phone everything works fine.

![screenshot from 2014-04-12 13 41 52](https://cloud.githubusercontent.com/assets/1431744/2687605/68fe8cf8-c261-11e3-97e0-f087910ce6f1.png)

**Comment by alvintc** [comment]
Yep, had the same thing on my N5 and now on my SGS5

Really annoying seeing the double messages, but atleast it's not double charging!

**Comment by DesiatoD** [comment]
Same issue. Moto X. Desksms also breaks long messages into pieces rather than sending them all at once. Very, very annoying as I use Desksms to send long messages that are slow to thumb type.

**Comment by JimiofEden** [comment]
I'm seeing this issue as well, SGS4, with the addition that it doesn't seem to send every other message. (i.e. it won't split them up and then won't send at all.)

**Comment by pauloschilling** [comment]
As I can see I've paid for another year of a broken app... :-1: 

@koush, you're active on GitHub as we can see on your profile page, is really hard to come here to help your users and customers? :(

**Comment by NZJourneyMan** [comment]
Hi I'm having the same issue. Ever since a recent phone os upgrade all txt messages are being sent twice. Another new behaviour is splitting longer messages (approx 300 characters) into two messages. Also very annoying as all phones I have used are happy to send messages of at least 8 sms's (never found an upper limit). 

I will _**not**_ be renewing until this fixed as I cannot use my browser to send txt messages. It's a shame that this has happened and doesn't seem to be getting fixed as I have been happily using desksms for years.

**Comment by DesiatoD** [comment]
Agreed with NZJourneyMan. This needs to get addressed or I'm out. Also agreed that it's a shame to see such a brilliant app get brought down by unaddressed bugs.

**Comment by JimiofEden** [comment]
So I'm not sure what has changed in the meantime, but my desksms is now working properly, without any known changes, and sending text messages.

Perhaps it was something to do with the way the carrier was handling the messages?

**Comment by NZJourneyMan** [comment]
I bailed, so haven't noticed the difference. What carrier are you using? I'm using Virgin Mobile which uses EE (T Mobile/Orange) as a backend.

**Comment by JimiofEden** [comment]
Verizon. It is splitting up the messages still, as a note, but it doesn't seem to stall at all or hold certain messages back.

**Comment by Alockmon** [comment]
This issue is still occurring about 9 months later. Nexus 4. I have installed every other SMS forwarding app like this and none do it as simply as DeskSMS but the double messages in the web app are a huge annoyance.

**Comment by CoolSilver** [comment]
I honestly stopped using this because of this bug months ago. It isn't just the Web app. Replying via Gmail and Tablet app also causes the duplicates.

**Comment by DesiatoD** [comment]
All right, it seems like it's time for this rat to jump ship. I'm giving Mightytext a shot as an alternative.

**Comment by NZJourneyMan** [comment]
I've switched to MobiText. It's ok, but I still prefer the DeskSMS interface :(
Let us know how Mightytext works out.

**Comment by koush** [comment]
Hi, I looked into this last night. It seems that on KitKat+, outgoing SMS messages are rewritten and there ends up being a timestamp mismatch. I'll see if I can work around this.;

**Comment by NZJourneyMan** [comment]
@koush, any luck?

**Comment by mrigali** [comment]
You've really got to do something about people renewing their subscriptions, paying the $5, not actually having the subscription take affect, and ignoring attempts at payment resolution. As such a respected member of the community, this reflects quite poorly on you, @koush.

**Comment by demangler** [comment]
Please fix the bug where duplicate messages are sent when using the web app. This renders it unusable. I just renewed my subscription because I thought maybe the latest Android (5) would fix it but it hasn't. 
Please fix it. Thanks
